src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -5,10 +5,19 @@
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int PASSWORD_MIN_LENGTH = 6;
+    private static final int AGE_MIN_VALUE = 18;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        if (user.getLogin().length() > 0
+                && storageDao.get(user.getLogin()) == null
+                && user.getAge() >= AGE_MIN_VALUE
+                && user.getPassword().length() >= PASSWORD_MIN_LENGTH) {
+            storageDao.add(user);
+            return user;
+        }
```

Comment:
```suggestion

```
what happen if user == null?

Reply:
Actually, I have nullUserTest, everything Ok, I receive RuntimeException. I agree, code does not look like clearly, but it works) Let's see what mentors will say.

Reply:
ok